src/systems/interactions.js

```diff
@@ -176,7 +176,13 @@ AFRAME.registerSystem("interaction", {
       !this.state.rightHand.held &&
       !this.rightHandTeleporter.isTeleporting &&
       !this.gazeTeleporter.isTeleporting;
-    this.cursorController.components["cursor-controller"].enabled = enableRightRemote;
+    const enableCursor =
+      !this.state.rightRemote.held ||
+      !this.state.rightRemote.held.components ||
+      !this.state.rightRemote.held.components.tags ||
+      !this.state.rightRemote.held.components.tags.data.isPen ||
+      this.state.rightRemote.held.children[0].components.pen.cursorVisible;
+    this.cursorController.components["cursor-controller"].enabled = enableRightRemote && enableCursor;
```

Comment:
I think it would be good to move the logic that determines `cursorVisible` into this line, so the whole predicate is made up of "the state of the world" at one point in time, rather than half of the predicate state being set when the pen component runs and the other half being determined here. Just seems